### DISCUSSION TITLE
WebView2 wants me to install WebView2

### DISCUSSION THREAD

**Discussion by Zenchess** [dolphinsmalltalk/Dolphin#1301] [Q&A] [answered]
wv := URLPresenter show: 'WebView2'.
wv value: 'https://github.com/DolphinSmalltalk/Dolphin'

The window that pops up says: 
"This content requires the Microsoft Edge WebView2 runtime". 

When I try to install either the x86 or x64 runtimes, it tells me they are already installed.  

I am using Dolphin 7.20, using the installer in 'Releases'

**Comment by Zenchess** [comment, marked as answer]
So, I managed to solve this.
I downloaded the nuget package for webview2 which you can find by googling 'webview2 sdk'. 
Opened the nuget package in winrar. 
Go to runtimes/win-x86/native
Put that file WebView2Loader.dll in your image folder

Now it works :)

**Reply by blairmcg** [reply]
Thank for identifying the issue and finding the resolution!

The loader is probably missing from the installer. I wouldn't know without checking, as I don't use it myself. I strongly recommend building the binaries and image from scratch, which is much easier and faster than it sounds. The readme has instructions. There are many advantages to doing this vs the installer:

1. The installer is untested. I don't use it. I don't test it.
2. The installer is essentially unmaintained. I don't want to spend time on it and nobody else is maintaining it.
3. An image built from the release/<version> branch (for 7.x) and master (for 8.x) will be up to date with latest fixes. The installer rarely is.
4. It is possible to run multiple minor versions side by side if you build them. I only bother changing the installer location for major versions. It is too much faff to do so every time there might be an incompatibility, and too expensive in development and testing time to provide forwards compatibility.
5. If you want to contribute a change, it is awkward to do so from an installer image.
6. If you need to debug anything at a low level, e.g. the integration to some third party code or component that is not written in Smalltalk, this will be much easier if you have built the binaries and image yourself.
7. No admin privileges are needed to build your own binaries and image. The installer makes machine wide changes which is not required if you build and run from a git clone. 
8. You may well get an OS security warning before running the installer after download that you have to override because it is not much downloaded/used. That makes even me nervous, even though I know roughly what it contains. This is just going to put people off, and understandably.
9. Building the binaries and image is easy and takes only a few minutes (the 8.x image boot runs in well under 30 seconds on my machine), and will give you a better understanding of how everything hangs together.

I could go on, but you get the idea. If it weren't for the fact that somebody in the past has gone to some trouble to put the installer script and build process together, I'd just drop it. If a significant issue with it arises and nobody comes forward to maintain it, then it will be cut.

**Reply by Zenchess** [reply]
Thanks for your response.  I've been using the installer for ages.  But I'll switch to building.

I do think the installer is still valuable to attract new users.  It's quite an ask to install visual studio just to try it out.

**Reply by blairmcg** [reply]
> I do think the installer is still valuable to attract new users. It's quite an ask to install visual studio just to try it out.

It's usually not necessary to build the binaries as the FetchVM script will download them from the release associated with the latest tag.

So the simplest set of steps is:
1) git clone
2) FetchVM
3) bootdpro
4) testdpro (optional)
5) dolphin7 (to start the booted image)

It would only be necessary to actually build the binaries if you want to debug, or for some reason the binaries are not available for download, e.g. because I forgot to tag after making a VM change. The readme does explain this, but its somewhat buried in the detail.
